datajunction-query/djqs/db/postgres.py

```diff
@@ -0,0 +1,138 @@
+"""
+Dependency for getting the postgres pool and running backend DB queries
+"""
+# pylint: disable=too-many-arguments
+from datetime import datetime
+from typing import List
+from uuid import UUID
+
+from fastapi import Request
+from psycopg import sql
+from psycopg_pool import AsyncConnectionPool
+
+from djqs.exceptions import DJDatabaseError
+
+
+async def get_postgres_pool(request: Request) -> AsyncConnectionPool:
+    """
+    Get the postgres pool from the app instance
+    """
+    app = request.app
+    return app.state.pool
+
+
+class DBQuery:
+    """
+    Metadata DB queries using the psycopg composition utility
+    """
+
+    def __init__(self):
+        self._reset()
+
+    def _reset(self):
+        self.selects: List = []
+        self.inserts: List = []
+
+    def get_query(self, query_id: UUID):
+        """
+        Get metadata about a query
+        """
+        self.selects.append(
+            sql.SQL(
+                """
+                SELECT id, catalog_name, engine_name, engine_version, submitted_query,
+                       async_, executed_query, scheduled, started, finished, state, progress
+                FROM query
+                WHERE id = {query_id}
+                """,
+            ).format(query_id=sql.Literal(query_id)),
+        )
+        return self
+
+    def save_query(
+        self,
+        query_id: UUID,
+        catalog_name: str = "",
+        engine_name: str = "",
+        engine_version: str = "",
+        submitted_query: str = "",
+        async_: bool = False,
+        state: str = "",
+        progress: float = 0.0,
+        executed_query: str = None,
+        scheduled: datetime = None,
+        started: datetime = None,
+        finished: datetime = None,
+    ):
+        """
+        Save metadata about a query
+        """
+        self.inserts.append(
```

Comment:
Can this just use the SQLAlchemy ORM directly, given that the models are already there and upserting a row like this would be trivial?